packages/core/src/extensions/Blocks/nodes/Block.ts

```diff
@@ -162,275 +109,408 @@ export const Block = Node.create<IBlock>({
+          "data-node-type": this.name,
         }),
         0,
       ],
     ];
   },
 
-  addInputRules() {
-    return [
-      ...["1", "2", "3"].map((level) => {
-        // Create a heading when starting with "#", "##", or "###""
-        return textblockTypeInputRuleSameNodeType({
-          find: new RegExp(`^(#{1,${level}})\\s$`),
-          type: this.type,
-          getAttributes: {
-            headingType: level,
-          },
-        });
-      }),
-      // Create a list when starting with "-"
-      textblockTypeInputRuleSameNodeType({
-        find: /^\s*([-+*])\s$/,
-        type: this.type,
-        getAttributes: {
-          listType: "li",
-        },
-      }),
-      textblockTypeInputRuleSameNodeType({
-        find: new RegExp(/^1.\s/),
-        type: this.type,
-        getAttributes: {
-          listType: "oli",
-        },
-      }),
-    ];
-  },
-
   addCommands() {
     return {
-      setBlockHeading:
-        (attributes) =>
-        ({ tr, dispatch }) => {
-          return setBlockHeading(tr, dispatch, attributes.level);
-        },
-      unsetBlockHeading:
-        () =>
-        ({ tr, dispatch }) => {
-          return setBlockHeading(tr, dispatch, undefined);
+      // Creates a new text block at a given position.
+      BNCreateBlock:
+        (pos) =>
+        ({ state }) => {
+          const newBlock = state.schema.nodes["block"].createAndFill()!;
+
+          state.tr.insert(pos, newBlock);
+
+          return true;
         },
-      unsetList:
-        () =>
-        ({ tr, dispatch }) => {
-          const node = tr.selection.$anchor.node(-1);
-          const nodePos = tr.selection.$anchor.posAtIndex(0, -1) - 1;
-
-          // const node2 = tr.doc.nodeAt(nodePos);
-          if (node.type.name === "block" && node.attrs["listType"]) {
-            if (dispatch) {
-              tr.setNodeMarkup(nodePos, undefined, {
-                ...node.attrs,
-                listType: undefined,
-              });
-              return true;
-            }
+      // Deletes a block at a given position and sets the selection to where the block was.
+      BNDeleteBlock:
+        (posInBlock) =>
+        ({ state, view }) => {
+          const blockInfo = getBlockInfoFromPos(state.doc, posInBlock);
+          if (blockInfo === undefined) {
+            return false;
           }
-          return false;
+
+          const { startPos, endPos } = blockInfo;
+
+          state.tr.deleteRange(startPos, endPos);
+          state.tr.setSelection(
+            new TextSelection(state.doc.resolve(startPos + 1))
+          );
+          view.focus();
+
+          return true;
         },
+      // Appends the text contents of a block to the nearest previous block, given a position between them. Children of
+      // the merged block are moved out of it first, rather than also being merged.
+      //
+      // In the example below, the position passed into the function is between Block1 and Block2.
```

Comment:
don't you mean betwee Block2 and Block3?

Reply:
It also doesn't work, compare with main branch when you backspace at start of  oldparent

<img width="356" alt="image" src="https://user-images.githubusercontent.com/368857/204822998-5725f0ee-7a75-44e3-9626-ed90d986b8dd.png">

Reply:
Yep, you're right, position should be between Block2 and Block3. Not sure why it's not working for you though, I tried it just now and the behavior for me is the same as on main. What does it do for you?

Reply:
hmm weird, works now. I'll let you know if I come across it again